xray/plot/plot.py

```diff
@@ -456,11 +472,16 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
                                  vmax=cmap_params['vmax'],
                                  **kwargs)
 
-        ax.set_xlabel(xlab)
-        ax.set_ylabel(ylab)
+        # Label the plot with metadata
+        if add_labels:
+            ax.set_xlabel(xlab)
+            ax.set_ylabel(ylab)
+            ax.set_title(_title_for_slice(darray))
 
         if add_colorbar:
-            plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
+            cbar = plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
+            if darray.name and add_labels:
```

Comment:
Makes sense. Just discovered this works: `a.name = (1, 2, 3)` but then fails to print. The `name` attribute is used to give the correct name with `to_dataset`. Do you know if it's used for other purposes also?

If I set an integer as a data variable name then the dataset won't save to netcdf.
So maybe it's worth restricting the name to be a string?